src/common/iconv.c

```diff
@@ -67,18 +67,27 @@ int guac_iconv(guac_iconv_read* reader, const char** input, int in_remaining,
     while (in_remaining > 0 && out_remaining > 0) {
 
         int value;
+        int count;
         const char* read_start;
         char* write_start;
 
         /* Read character */
         read_start = *input;
         value = reader(input, in_remaining);
-        in_remaining -= *input - read_start;
+        count = *input - read_start;
+        /* Stop if not enough bytes to read */
+        if (count == 0)
+            return 1;
+        in_remaining -= count;
 
         /* Write character */
         write_start = *output;
         writer(output, out_remaining, value);
-        out_remaining -= *output - write_start;
+        count = *output - write_start;
+        /* Stop if not enough room */
```

Comment:
Please include a blank line before comments. The comment (and subsequent code block) becomes difficult to distinguish from the rest of the code otherwise.